### ISSUE TITLE
[BUG] Template().construct plotdir issue on windows10

### ISSUE THREAD

**Issue by Halauwet** [eqcorrscan/EQcorrscan#344]
Every time I construct template with `plotdir` or `save_progress` always get [Errno 22].
I've changed the target-directory slash to backslash or double slash/backslash still have the same error.

Here is the error detail.
```python
OSError                                   Traceback (most recent call last)
<ipython-input-63-67b2f9bcd4da> in <module>
     20 template = Template().construct(method='from_sac', name='ambon5.6', lowcut=1.0, highcut=8.0, samp_rate=20.0, \
     21             filt_order=4, length=15, prepick=0.2, swin='all', skip_short_chans=True, \
---> 22             delayed=True, plot=True, plotdir=(parentdir.replace('\\','/') + '/' + templatedir), sac_files=waveforms)
     23 
     24 

~\Anaconda3\envs\eqcorrscan\lib\site-packages\eqcorrscan\core\match_filter\template.py in construct(self, method, name, lowcut, highcut, samp_rate, filt_order, length, prepick, swin, process_len, all_horiz, delayed, plot, plotdir, min_snr, parallel, num_cores, skip_short_chans, **kwargs)
    646             all_horiz=all_horiz, delayed=delayed, plot=plot, plotdir=plotdir,
    647             min_snr=min_snr, parallel=parallel, num_cores=num_cores,
--> 648             skip_short_chans=skip_short_chans, **kwargs)
    649         self.name = name
    650         st = streams[0]

~\Anaconda3\envs\eqcorrscan\lib\site-packages\eqcorrscan\core\template_gen.py in template_gen(method, lowcut, highcut, samp_rate, filt_order, length, prepick, swin, process_len, all_horiz, delayed, plot, plotdir, return_event, min_snr, parallel, num_cores, save_progress, skip_short_chans, **kwargs)
    395                 event.picks, st, length, swin, prepick=prepick, plot=plot,
    396                 all_horiz=all_horiz, delayed=delayed, min_snr=min_snr,
--> 397                 plotdir=plotdir)
    398             process_lengths.append(len(st[0].data) / samp_rate)
    399             temp_list.append(template)

~\Anaconda3\envs\eqcorrscan\lib\site-packages\eqcorrscan\core\template_gen.py in _template_gen(picks, st, length, swin, prepick, all_horiz, delayed, plot, min_snr, plotdir)
    830               savefile="{0}/{1}_template.png".format(
    831                   plotdir, st1[0].stats.starttime),
--> 832               **plot_kwargs)
    833         noise_plot(signal=st1, noise=noise,
    834                    savefile="{0}/{1}_noise.png".format(

~\Anaconda3\envs\eqcorrscan\lib\site-packages\eqcorrscan\utils\plotting.py in pretty_template_plot(template, background, picks, **kwargs)
   1439     plt.tight_layout()
   1440     plt.subplots_adjust(hspace=0)
-> 1441     fig = _finalise_figure(fig=fig, **kwargs)  # pragma: no cover
   1442     return fig
   1443 

~\Anaconda3\envs\eqcorrscan\lib\site-packages\eqcorrscan\utils\plotting.py in _finalise_figure(fig, **kwargs)
     77         fig.suptitle(title)
     78     if save:
---> 79         fig.savefig(savefile, bbox_inches="tight")
     80         Logger.info("Saved figure to {0}".format(savefile))
     81     if show:

~\Anaconda3\envs\eqcorrscan\lib\site-packages\matplotlib\figure.py in savefig(self, fname, **kwargs)
   2060             self.set_frameon(frameon)
   2061 
-> 2062         self.canvas.print_figure(fname, **kwargs)
   2063 
   2064         if frameon:

~\Anaconda3\envs\eqcorrscan\lib\site-packages\matplotlib\backend_bases.py in print_figure(self, filename, dpi, facecolor, edgecolor, orientation, format, **kwargs)
   2261                 orientation=orientation,
   2262                 bbox_inches_restore=_bbox_inches_restore,
-> 2263                 **kwargs)
   2264         finally:
   2265             if bbox_inches and restore_bbox:

~\Anaconda3\envs\eqcorrscan\lib\site-packages\matplotlib\backends\backend_agg.py in print_png(self, filename_or_obj, *args, **kwargs)
    528 
    529         try:
--> 530             with cbook.open_file_cm(filename_or_obj, "wb") as fh:
    531                 _png.write_png(renderer._renderer, fh,
    532                                self.figure.dpi, metadata=metadata)

~\Anaconda3\envs\eqcorrscan\lib\contextlib.py in __enter__(self)
    110         del self.args, self.kwds, self.func
    111         try:
--> 112             return next(self.gen)
    113         except StopIteration:
    114             raise RuntimeError("generator didn't yield") from None

~\Anaconda3\envs\eqcorrscan\lib\site-packages\matplotlib\cbook\__init__.py in open_file_cm(path_or_file, mode, encoding)
    624 def open_file_cm(path_or_file, mode="r", encoding=None):
    625     r"""Pass through file objects and context-manage `.PathLike`\s."""
--> 626     fh, opened = to_filehandle(path_or_file, mode, True, encoding)
    627     if opened:
    628         with fh:

~\Anaconda3\envs\eqcorrscan\lib\site-packages\matplotlib\cbook\__init__.py in to_filehandle(fname, flag, return_opened, encoding)
    609             fh = bz2.BZ2File(fname, flag)
    610         else:
--> 611             fh = io.open(fname, flag, encoding=encoding)
    612         opened = True
    613     elif hasattr(fname, 'seek'):

OSError: [Errno 22] Invalid argument: 'D:/project/obspy/EQdetection/templates/2019-09-26T00:39:57.037500Z_template.png'
```

**Comment by calum-chamberlain** [comment]
Morning!  Sorry about this - it looks like a Windows specific error - a quick google highlights that `:` is a special character that likely Windows isn't allowing in the filename.  Unfortunately this needs to get changed internally to EQcorrscan.  If you want to you can try changing line 831 of template_gen.py so that the second argument (`st1[0].stats.starttime`) is formatted without :, e.g. change it to something like:
`st1[0].stats.starttime.strftime("%Y-%m-%dT%H-%M-%S")`

**Comment by Halauwet** [comment]
thanks @calum-chamberlain it's solve my problem.

I also changed line 406 and 835 for formated-filename of mseed_progress and noise_plot.

**Comment by calum-chamberlain** [comment]
Would you be willing to submit a pull request with your changes please? Then we can fix the bug.

Thanks!

CJ Chamberlain, out of office

________________________________

**Comment by calum-chamberlain** [comment]
Closed by #351
